# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/nsICookie.idl
# Commit: 6379a4984eba
# Full Hash: 6379a4984ebafb659428ec3e4493f72173b3a70a
# Author: Paul Zuehlcke <pbz@mozilla.com>
# Date: 2023-09-22 04:22:03
# Description:
#   Bug 1854028 - Update BounceTrackingStorageObserver cookie check and remove nsICookieNotification isThirdPartyCookie. r=bvandersloot,cookie-reviewers,anti-tracking-reviewers,timhuang
#   
#   For the BounceTrackingStorageObserver, instead of passing in a third-party flag
#   we can simply check if the changed cookie is partitioned or not.
#   
# ==============================================================================

diff -r 24c5f04d3793 -r 6379a4984eba netwerk/cookie/nsICookie.idl
--- a/netwerk/cookie/nsICookie.idl	Thu Sep 21 14:30:17 2023 +0000
+++ b/netwerk/cookie/nsICookie.idl	Thu Sep 21 14:42:32 2023 +0000
@@ -11,11 +11,16 @@
  */
 
 %{C++
-namespace mozilla::net {
-class Cookie;
+namespace mozilla {
+  class OriginAttributes;
+
+  namespace net {
+    class Cookie;
+  }
 }
 %}
 
+[ref] native const_OriginAttributes(const mozilla::OriginAttributes);
 [ref] native const_Cookie(const mozilla::net::Cookie);
 
 typedef long nsCookieStatus;
@@ -88,6 +93,12 @@
     [implicit_jscontext]
     readonly attribute jsval originAttributes;
 
+    /**
+     * Native getter for origin attributes
+     */
+    [noscript, notxpcom, nostdcall, binaryname(OriginAttributesNative)]
+    const_OriginAttributes OriginAttributesNative();
+
     [noscript, notxpcom, nostdcall, binaryname(AsCookie)]
     const_Cookie AsCookie();
 